### DISCUSSION TITLE
Returning queryset instead of list in paginate_queryset()

### DISCUSSION THREAD

**Discussion by revanthgss** [encode/django-rest-framework#8264] [Ideas & Suggestions]
The paginate_queryset() function in PageNumberPagination Class is returning list(self.page) instead of self.page. Using list() evalutates the queryset and if we need to do further processing, we need to make another query. If we use queryset, the page can be directly fetched in list serializers and processed further without evaluating queryset. 

I think the list is used because some of the custom paginations like CursorPagination might force us to evaluate the queryset.

Please put down your thoughts and suggestions regarding this.

**Comment by ltshb** [comment]
I've a similar problem with the CursorPagitation. After the pagination I need to make an aggregation query on the paginated results in order to the correct cache-control header (depending on the paginated result I might set higher/lower cache-control max-age). It would great if the `paginate_queryset()` method would return also the paginated queryset or save as member in order to reuse the un-evaluated paginated queryset for other purpose.
